core/trino-spi/src/main/java/io/trino/spi/block/DictionaryBlock.java

```diff
@@ -273,6 +273,12 @@ public long getLogicalSizeInBytes()
             return logicalSizeInBytes;
         }
 
+        OptionalInt dictionarySizePerPosition = dictionary.fixedSizeInBytesPerPosition();
```

Comment:
From `getLogicalSizeInBytes()`'s documentation:
`Returns the size of the block contents, regardless of internal representation. The same logical data values should always have the same size, no matter what block type is used or how they are represented within a specific block.`
AFAIU `fixedSizeInBytesPerPosition` may contain block overhead size. Is that a problem?

Reply:
I looked at implementations of `fixedSizeInBytesPerPosition`. They all take into account the size of the type, size of boolean for nulls array and the size of an integer for offsets array in case of row block. The `getRegionSizeInBytes` implementations used here also take into account the same things.
Did you see anything in particular that would indicate that fixedSizeInBytesPerPosition may calculate size in way that is different from the current implementation ?

Reply:
no